nav2_costmap_2d/src/costmap_topic_collision_checker.cpp

```diff
@@ -96,25 +99,27 @@ double CostmapTopicCollisionChecker::scorePose(
 Footprint CostmapTopicCollisionChecker::getFootprint(const geometry_msgs::msg::Pose2D & pose)
 {
   Footprint footprint;
-  if (!footprint_sub_.getFootprint(footprint)) {
+  rclcpp::Time stamp = clock_->now();
+  if (!footprint_sub_.getFootprint(footprint, stamp)) {
```

Comment:
All https://github.com/ros-planning/navigation2/blob/cf10abebfee60fb5cf09df6836b24594b68c083c/nav2_costmap_2d/src/footprint_subscriber.cpp#L94 does is call https://github.com/ros-planning/navigation2/blob/cf10abebfee60fb5cf09df6836b24594b68c083c/nav2_costmap_2d/src/footprint_subscriber.cpp#L85 which has the `now()` call in it, why change the use here? Does this buy us something? This looks like the usecase for your new `getFootprint()` function, but I'm not seeing how this helps in synchronizing, it looks like its just shuffling those lines around rather than using an existing API

Reply:
Oh I see, its so you can share the same `now()` timestamp with `unorientFootprint()`. Is there anywhere where the old just `getFootprint(footprint)` API is still used and wouldn't be able to be updated to use the new one specifying `now()` as an argument? If not, delete that old function and replace all its uses with this one.

Reply:
No, it's not the same `now()`, `stamp` is being updated inside `getFootprint(footprint, stamp)`. After 3 unsuccessful attempts of explaining these few lines I think it's time to do a proper cleanup of FootprintSubscriber and remove everything that's misleading and useless (namely `footprint_timeout` check) + to write a proper doxygen, so that FootprintSubscriber is less confusing and easy to understand.

Reply:
I agree, this was from an old Intel design when they were trying to have an independent environmental model and use the footprint topic as more of a word-of-god type thing synchronized with the costmaps for use across the system. This is code we haven't seriously looked at in about 2 years because it was refactored out, now only used by the recovery behaviors which weren't as context sensitive. 

For context, the original design intent was to make sure that if the footprint wasn't received in synchronous with the costmap (before the topic sychronizers were ported to ROS 2) that we'd fail planning or control because a dropped footprint message could be fatal in a bad situation if assumed to be paired with a costmap update. But now that we don't use this tool anymore in that context, the timeout is nonsensical as its morphed to have a refinement process to get it into the correct frame in an updated timestamp.